indico/web/client/js/custom_elements/ind_collapsible.js

```diff
@@ -0,0 +1,79 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import {domReady} from 'indico/utils/domstate';
+
+import './ind_collapsible.scss';
+
+customElements.define(
+  'ind-collapsible',
+  class extends HTMLElement {
+    connectedCallback() {
+      domReady.then(() => {
+        this.container = this.querySelector('[data-collapsible]');
+        this.containerStyle = window.getComputedStyle(this.container);
+
+        console.assert(
+          this.container,
+          '<ind-collapsible> must have a descendant with a data-collapsible attribute'
+        );
+        console.assert(
+          this.querySelector('button[value="toggle-collapse"]'),
+          '<ind-collapsible> must have a descendant button with value="toggle-collapse"'
+        );
+
+        this.markCollapsible();
+
+        window.addEventListener('resize', () => this.markCollapsible());
+        window.addEventListener('orientationchange', () => this.markCollapsible());
+        this.addEventListener('click', evt => {
+          if (evt.target.closest('button[value="toggle-collapse"]')) {
+            this.toggleAttribute('collapsed');
+          }
+        });
+      });
+    }
+
+    disconnectedCallback() {
+      window.removeEventListener('resize', this.markCollapsible);
+      window.removeEventListener('orientationchange', this.markCollapsible);
+    }
```

Comment:
Will this actually remove the event listeners? The function passed to `addEventListener` is different from the one passed to `removeEventListener`.